cli/src/cli.ts

```diff
@@ -31,10 +31,13 @@ async function runUser(
   return logs;
 }
 
-async function main (task: Task, numberOfUsers: number): Promise<void> {
+async function main (provider: TaskProvider, numberOfUsers: number): Promise<void> {
+  const task = provider.getTask()
   console.log(`Started ${task.trainingInformation.scheme} training of ${task.id}`)
   console.log({ args })
-  const [server, url] = await startServer()
+
+  const disco = await Server.of(provider)
+  const [server, url] = await disco.serve()
```

Comment:
```suggestion
  const discoServer = await Server.of(provider)
  const [server, url] = await discoServer.serve()
```
What about renaming the variable names to not use the same one as in `runUser` (where `const disco` is a `Disco` object, not `Server`)

Reply:
good idea, that's more explicit